test_agent/rust/src/testagent.rs

```diff
@@ -0,0 +1,356 @@
+            &[("payload".to_string(), value_str.to_string())]
+                .iter()
+                .cloned()
+                .collect::<HashMap<_, _>>(),
+        ) else {
+            error!("Issue in converting to payload");
+            return;
+        };
+
+        let data = [("data".to_string(), payload_str.to_string())]
+            .iter()
+            .cloned()
+            .collect::<HashMap<_, _>>();
+        let json_message = JsonResponseData {
+            action: constants::RESPONSE_ON_RECEIVE.to_owned(),
+            data,
+            ue: "rust".to_string(),
+            test_id: "1".to_string(),
+        };
+
+        dbg!("sending received data to tm....");
+        let json_message_str = convert_json_to_jsonstring(&json_message);
+        let message = json_message_str.as_bytes();
+
+        let Ok(mut socket) = self.clientsocket_to_tm.try_lock() else {
+            error!("Failed to acquire lock for sending data to TM from on_receive");
+            return;
+        };
+
+        let result = socket.write_all(message);
+        match result {
+            Ok(()) => println!("on receive could send data to TM"),
+            Err(err) => error!("on receive could not send data to TM{}", err),
+        }
+    }
+
+    async fn on_error(&self, _err: UStatus) {
+        dbg!(_err);
+    }
+}
+
+impl SocketTestAgent {
+    pub fn new(test_clientsocket: TcpStream, listener: Arc<dyn UListener>) -> Self {
+        let clientsocket = Arc::new(Mutex::new(test_clientsocket));
+
+        Self {
+            clientsocket,
+            listener,
+        }
+    }
+
+    async fn handle_send_command(
+        &self,
+        utransport: &UTransportSocket,
+        json_data_value: Value,
+    ) -> Result<(), UStatus> {
+        let wrapper_umessage: WrapperUMessage = match serde_json::from_value(json_data_value) {
+            Ok(message) => message,
+            Err(err) => {
+                error!("not able to deserialize send UMessage from Json{}", err);
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    "Failed to Deserialize",
+                ));
+            }
+        };
+        let u_message = wrapper_umessage.0;
+        utransport.send(u_message).await
+    }
+
+    async fn handle_register_listener_command(
+        &self,
+        utransport: &UTransportSocket,
+        json_data_value: Value,
+    ) -> Result<(), UStatus> {
+        let wrapper_uuri: WrapperUUri = match serde_json::from_value(json_data_value) {
+            Ok(message) => message,
+            Err(err) => {
+                error!(
+                    "not able to deserialize register listener UUri from Json{}",
+                    err
+                );
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    "Failed to Deserialize",
+                ));
+            }
+        };
+        let u_uuri = wrapper_uuri.0;
```

Comment:
Same sort of comment as [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1597095275).